### ISSUE TITLE
3.0 Shading language suggestion: vec2, vec3, and vec4 overloads for trig functions

### ISSUE THREAD

**Issue by Smellyhobo101** [godotengine/godot#10562]
It looks like there are vec2, vec3, and vec4 versions of most of the built in functions in the 3.0 shading language, but not for any of the trig functions:

https://docs.google.com/spreadsheets/d/1hBpkNor5-WUTKuiEP3Myo4wzleCWvqPqn1OLgOwFsmU/edit#gid=1952867603

Would make porting GLSL code easier. For example [this tutorial](https://thebookofshaders.com/12/) on
cell noise has a random function that uses a vec2 sin():

```
vec2 random2( vec2 p ) {
    return fract(sin(vec2(dot(p,vec2(127.1,311.7)),dot(p,vec2(269.5,183.3))))*43758.5453);
}
```

**Comment by Zephilinox** [comment]
This was fixed with #13294 right?

**Comment by reduz** [comment]
yup
